src/couch_views/src/couch_views_indexer.erl

```diff
@@ -87,7 +88,49 @@ init() ->
         design_opts => Mrst#mrst.design_opts
     },
 
-    update(Db, Mrst, State).
+    try
+        update(Db, Mrst, State)
+    catch
+        exit:normal ->
+            ok;
+        Error:Reason  ->
+            NewRetry = Retries + 1,
+            RetryLimit = retry_limit(),
+
+            if NewRetry < RetryLimit -> ok; true ->
+                    NewData = add_error(Error, Reason, Data),
+                    couch_jobs:finish(undefined, Job, NewData),
+                    exit(normal)
+            end,
+
+            DataErr = Data#{<<"retries">> := NewRetry},
+            StateErr = State#{job_data := DataErr, last_seq := <<"0">>},
+            report_progress(StateErr, update)
+    end.
+
+add_error(error, {erlfdb_error, 2101}, Data) ->
+    Data#{
+        error => transaction_too_large,
+        reason => <<"Transaction exceeds byte limit">>
+    };
+
+add_error(error, {erlfdb_error, 1031}, Data) ->
+    Data#{
+        error => transaction_timed_out,
+        reason => <<"Operation aborted because the transaction timed out">>
+    };
+
+add_error(error, {erlfdb_error, Code}, Data) ->
+    Data#{
+        error => foundationdb_error,
+        reason => integer_to_binary(Code)
+    };
+
+add_error(Error, Reason, Data) ->
+    Data#{
+        error => couch_util:to_binary(Error),
+        reason => couch_util:to_binary(Reason)
+    }.
```

Comment:
Here we could use the now exposed get_error_string API function from erlfdb to get the description:

https://github.com/cloudant-labs/couchdb-erlfdb/pull/9

If we wanted to keep the code, could use a list, since it's a valid json data structure.

```
error => [foundationdb_error, Code],
reason => erlfdb:get_error_string(Code)
```